File: proselint/checks/misc/apologizing.py
"""
Excessive apologizing.

---
layout:     post
source:     Pinker's book on writing
source_url: ???
title:      excessive apologizing
date:       2014-06-10 12:31:19
categories: writing
---

Points out excessive apologizing.

"""

from proselint.registry.checks import Check, types

check = Check(
    check_type=types.ExistenceSimple(pattern="More research is needed"),
    path="misc.apologizing",
    message="Excessive apologizing.",
)

__register__ = (check,)

File: proselint-rs/crates/proselint-checks/src/misc/apologizing.rs
use proselint_registry::{pad, checks::{Check, CheckType, Padding}};

pub const EXAMPLES_PASS: &[&str] = &["Smoke phrase with nothing flagged."];
pub const EXAMPLES_FAIL: &[&str] = &["To say more research is needed."];

const CHECK: Check = Check {
	check_type: CheckType::ExistenceSimple {
		pattern: pad!(Padding::WordsInText, "more research is needed"),
		unicode: true,
		exceptions: &[],
	},
	path: "misc.apologizing",
	msg: "Excessive apologizing.",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];
